File: Exercises/geoangelotti/unique_count.go
package epwc

import (
	"slices"
	"sort"
)

func UniqueCount1(data []int) int {
	sort.Slice(data, func(i, j int) bool {
		return data[i] < data[j]
	})
	if len(data) == 0 {
		return 0
	}
	count := 1
	for i := 1; i < len(data); i++ {
		if data[i] != data[i-1] {
			count++
		}
	}
	return count
}

func UniqueCount2(data []int) int {
	slices.Sort(data)
	if len(data) == 0 {
		return 0
	}
	count := 1
	for i := 1; i < len(data); i++ {
		if data[i] != data[i-1] {
			count++
		}
	}
	return count
}

func UniqueCount3(data []int) int {
	uniques := make(map[int]bool)
	for _, num := range data {
		if !uniques[num] {
			uniques[num] = true
		}
	}
	return len(uniques)
}

File: Exercises/geoangelotti/unique_count.jl
function unique_count(data::Vector{Int32})::Int
    sorted_data = sort(data)
    count = 0
    if !isempty(sorted_data)
        count = 1
        for i in 2:length(sorted_data)
            if sorted_data[i] != sorted_data[i-1]
                count += 1
            end
        end
    end
    return count
end

function main()
    arr = Int32[1, 3, 1, 4, 1, 5]
    println(unique_count(arr))
end

main()